tests/data_context/test_base_data_context.py

```diff
@@ -264,10 +283,11 @@ def test_get_validator_with_cloud_enabled_context_saves_expectation_suite_to_clo
     """
     context = DataContext(ge_cloud_mode=True)
 
+    prep = prepare_validator_for_cloud_e2e(context)
     (
         validator,
         _,
-    ) = prepare_validator_for_cloud_e2e(context)
+    ) = next(prep)
```

Comment:
Shouldn't need to use `next()` here.
The usage/invocation of the fixture shouldn't need to change.

The closure should probably remain a regular callable.
The fixture itself should `yield` the closure and then the teardown logic should be defined after the yield
```python
def prepare_validator_for_cloud_e2e():
   ....
   yield _closure
   # teardown logic goes here
```